### PR TITLE
ADEN-5602 Add MEGA ad unit builder to mobile FV pages

### PR DESCRIPTION
**Pull request by v3rth**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdContext.spec.js](https://github.com/Wikia/app/blob/90e8c227b77d9caf49a02fd9507de596b27b5723/extensions/wikia/AdEngine/js/spec/AdContext.spec.js): File "AdContext.spec.js" has 1014 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/AdEngine/js/context/adContext.js [hunk 1]
```diff
@@ -91,25 +91,37 @@ define('ext.wikia.adEngine.adContext', [
...
 				geo.isProperGeo(instantGlobals.wgAdDriverPremiumAdLayoutCountries)
 			);
 
-		context.opts.premiumAdLayoutEnabled = isEnabledOnFeaturedVideo || isEnabledOnRegularArticle;
+		context.opts.premiumAdLayoutEnabled = isPALEnabledOnDesktopFeaturedVideo || isPALEnabledOnRegularArticle;
 		context.slots.premiumAdLayoutSlotsToUnblock = ['INCONTENT_BOXAD_1', 'BOTTOM_LEADERBOARD'];
 
-		context.opts.megaAdUnitBuilderEnabled = context.targeting.hasFeaturedVideo && isEnabledOnFeaturedVideo &&
-			geo.isProperGeo(instantGlobals.wgAdDriverMegaAdUnitBuilderForFVCountries);
+		context.opts.megaAdUnitBuilderEnabled = isMEGAEnablefForFVOasis(context, isPALEnabledOnDesktopFeaturedVideo) ||
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/context/adContext.js]: Hm, is MEGA really related to PAL? `hasFeaturedVideo` + wgCountries wouldn't be sufficient?

File: extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js [hunk 2]
```diff
@@ -67,12 +67,16 @@ define('ext.wikia.adEngine.provider.factory.wikiaGpt', [
 			});
 		}
 
+		function getExtraAdUnitBuilder() {
+			return typeof extra.adUnitBuilder === 'function' ? extra.adUnitBuilder() : extra.adUnitBuilder;
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js]: Wouldn't it be easier to change all `extra.adUnitBuilder` to  `extra.getAdUnitBuilder` and use it as function everywhere?

File: extensions/wikia/AdEngine/js/context/adContext.js (hunk 1 above)
Comment (v3rth) [reply extensions/wikia/AdEngine/js/context/adContext.js]: It looks like a business decision, if we want to be able to turn it on for non-PAL FV pages it shouldn't be related

File: extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js (hunk 2 above)
Comment (v3rth) [reply extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js]: Nope, this is easier :grin: but probably not cleaner...

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdContext.spec.js](https://github.com/Wikia/app/blob/9081b51241afeccab5215618fe09595c11a58c0f/extensions/wikia/AdEngine/js/spec/AdContext.spec.js): File "AdContext.spec.js" has 1014 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdContext.spec.js](https://github.com/Wikia/app/blob/f2600713a6538c4cb42a11cb3d6e1b2571dbafdb/extensions/wikia/AdEngine/js/spec/AdContext.spec.js): File "AdContext.spec.js" has 1014 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdContext.spec.js](https://github.com/Wikia/app/blob/039f221ed2a7dba7efeb943281ba49886eab557f/extensions/wikia/AdEngine/js/spec/AdContext.spec.js): File "AdContext.spec.js" has 1014 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by fraszczakszymon — APPROVED** [review]

**Pull request merged**